src/js/services/opencap.service.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+(function() {
+  angular.module('bitcoincom.services').factory('opencapService', function($q, $http) {
+    function getAddress(alias) {
+      let aliasData = validateAlias(alias);
+      if (aliasData.username === '' || aliasData.domain === '') {
+        return $q((resolve, reject) => {
+          return reject('Invalid OpenCAP alias');
+        });
+      }
+
+      let deferred = $q.defer();
+      $http
+        .get(`https://dns.google.com/resolve?name=_opencap._tcp.${aliasData.domain}&type=SRV`)
+        .then(function(response) {
+          deferred.resolve(
+            parseSRV(response.data)
+              .then(data => getAddresses(alias, data.host, data.dnssec))
+              .catch(function(error) {
+                return $q((resolve, reject) => {
+                  reject(error);
+                });
+              })
+          );
+        })
+        .catch(function(response) {
+          deferred.reject('Couldn\'t find srv record for the provided domain');
+        });
+      return deferred.promise;
+    }
+
+    var parseSRV = function(respData) {
+      return $q((resolve, reject) => {
+        let dnssec = respData.AD;
+
+        if (typeof respData.Answer === 'undefined') {
+          return reject('Error contacting google dns server, no srv data');
+        }
+        if (respData.Answer.length < 1) {
+          return reject('Error contacting google dns server, not enough srv data');
+        }
+
+        let record = respData.Answer[0].data.split(' ');
+        if (record.length != 4) {
+          return reject('Error contacting google dns server, improper srv data');
+        }
+
+        if (record[3].slice(-1) == '.') {
+          record[3] = record[3].substring(0, record[3].length - 1);
+        }
+
+        return resolve({ host: record[3], dnssec });
+      });
+    };
+
+    var getAddresses = function(alias, host, dnssec) {
+      let deferred = $q.defer();
+      $http
+        .get(`https://${host}/v1/addresses?alias=${alias}`)
+        .then(function(response) {
+          deferred.resolve(parseAddresses(response.data, dnssec).then());
+        })
+        .catch(function(response) {
+          deferred.reject('Address not found for the specified alias');
+        });
+      return deferred.promise;
+    };
+
+    var parseAddresses = function(respData, dnssec) {
+      let addresses = {}
+      return $q((resolve, reject) => {
```

Comment:
Now :  `=>`  : lambda expression is not supported for old javascript engine.
Should be : 
```
function onQ() {
}
```